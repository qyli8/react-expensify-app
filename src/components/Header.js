import React from "react";
import { NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {logout} from '../actions/auth'
import {history} from '../routes/AppRouter'
export const Header = (props) =>{
  const logoutAccount=()=>{
    history.push('/')
    props.logout('UNAUTHORIZED')
  }
  return(
    <header>
      <h1>Expensify</h1>
      <NavLink to="/home" activeClassName="is-active" exact={true}>DashBoard</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create</NavLink>
      <NavLink to="/help" activeClassName='is-active'>Help</NavLink>
      <button  onClick={logoutAccount}>Logout</button>
    </header>
  )
}

const mapDispatchToProps = (dispatch)=>({
  logout: (status)=>dispatch(logout(status))
})

export default connect(undefined, mapDispatchToProps)(Header);