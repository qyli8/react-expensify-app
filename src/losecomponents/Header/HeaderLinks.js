/*eslint-disable*/
import React from "react";
import {clearSession} from '../../actions/auth'
import {connect} from 'react-redux'

import {history} from '../../routes/AppRouter'
// react components for routing our app without refresh
import { Link } from "react-router-dom";

// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import Tooltip from "@material-ui/core/Tooltip";

// @material-ui/icons
import  Apps from "@material-ui/icons/Apps";
import ScreenLockPortrait  from "@material-ui/icons/ScreenLockPortrait"

// core components
import CustomDropdown from "../CustomDropdown/CustomDropdown.js";
import Button from "../CustomButtons/Button.js";

import styles from "../../assets/jss/material-kit-react/components/headerLinksStyle.js";

const useStyles = makeStyles(styles);

export const HeaderLinks =(props)=> {
  const logoutAccount=()=>{
    history.push('/')
    props.logout('UNAUTHORIZED')
    console.log()
  }
  const classes = useStyles();
  return (
    <List className={classes.list}>
      {/* <ListItem className={classes.listItem}>
        <CustomDropdown
          noLiPadding
          buttonText="Components"
          buttonProps={{
            className: classes.navLink,
            color: "transparent"
          }}
          // buttonIcon={Apps}
          dropdownList={[
            <Link to="/" className={classes.dropdownLink}>
              All components
            </Link>,
            <a
              href="https://creativetimofficial.github.io/material-kit-react/#/documentation?ref=mkr-navbar"
              target="_blank"
              className={classes.dropdownLink}
            >
              Documentation
            </a>
          ]}
        />
      </ListItem> */}
      <ListItem className={classes.listItem}>
        <Button
          onClick={logoutAccount}
          color="transparent"
          className={classes.navLink}
        >
          <ScreenLockPortrait className={classes.icons} /> Logout
        </Button>
      </ListItem>
      {/* <ListItem className={classes.listItem}>
        <Tooltip title="Delete">
          <IconButton aria-label="Delete">
            <DeleteIcon />
          </IconButton>
        </Tooltip>
        <Tooltip
          id="instagram-twitter"
          title="Follow us on twitter"
          placement={window.innerWidth > 959 ? "top" : "left"}
          classes={{ tooltip: classes.tooltip }}
        >
          <Button
            href="https://twitter.com/CreativeTim?ref=creativetim"
            target="_blank"
            color="transparent"
            className={classes.navLink}
          >
            <i className={classes.socialIcons + " fab fa-twitter"} />
          </Button>
        </Tooltip>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Tooltip
          id="instagram-facebook"
          title="Follow us on facebook"
          placement={window.innerWidth > 959 ? "top" : "left"}
          classes={{ tooltip: classes.tooltip }}
        >
          <Button
            color="transparent"
            href="https://www.facebook.com/CreativeTim?ref=creativetim"
            target="_blank"
            className={classes.navLink}
          >
            <i className={classes.socialIcons + " fab fa-facebook"} />
          </Button>
        </Tooltip>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Tooltip
          id="instagram-tooltip"
          title="Follow us on instagram"
          placement={window.innerWidth > 959 ? "top" : "left"}
          classes={{ tooltip: classes.tooltip }}
        >
          <Button
            color="transparent"
            href="https://www.instagram.com/CreativeTimOfficial?ref=creativetim"
            target="_blank"
            className={classes.navLink}
          >
            <i className={classes.socialIcons + " fab fa-instagram"} />
          </Button>
        </Tooltip>
      </ListItem> */}
    </List>
  );
}
const mapDispatchToProps =(dispatch)=>({
  logout: (status)=>dispatch(clearSession(status))
})

export default connect(undefined,mapDispatchToProps)(HeaderLinks);