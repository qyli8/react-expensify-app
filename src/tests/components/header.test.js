import React from 'react'
import {shallow} from 'enzyme';
import Header from '../../components/Header'

test('should render Header corretly', ()=>{
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();

  // const renderer = new ShallowRenderer();
  // renderer.render(<Header />)
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  // console.log(renderer.getRenderOutput())

})