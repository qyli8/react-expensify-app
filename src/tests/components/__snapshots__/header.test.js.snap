// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header corretly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header_content"
    >
      <[object Object]
        className="header_title"
        to="/home"
      >
        <h1>
          Expensify
        </h1>
      </[object Object]>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
